### PR TITLE
Contribution guide

### LINKED ISSUE #304: [DOC] Add CONTRIBUTING.md
### 📃 Documentation

Add a guideline to contributors.

### Motivation

Clarify steps that need to be fulfilled before a PR get's approved:

- Create an issue or work on an existing one.
- Use recommended install to avoid python version mismatches
- Formatting and Style:`make style` needs to pass
- Tests: need to pass (guideline how they can be run locally)
- ...

### PR DESCRIPTION
**Pull request by maxjeblick**
This PR adds contribution guidelines for Llm Studio. Those apply both to opening a new issue, as well as opening a PR.

@sherenem : Could you provide a section describing the workflow on how to add/change documentation?

Fixes #304

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+   your contribution.
+
+## Setting up your development environment
+
+Follow the instructions in [README](https://github.com/h2oai/h2o-llmstudio/blob/main/README.md) to set up your
+environment. Run `make setup-dev` instead of `make setup` to install the development dependencies.
+
+## Running linters and tests
+
+Before submitting your pull request, ensure that your code adheres to our guidelines and passes the linters and tests.
```
Comment (sherenem) [inline CONTRIBUTING.md:43]: Do we have a link to point off to for more info on what guidelines the code should adhere to?

**Comment by sherenem** [comment]
Ack. I've pointed off to the [documentation readme](https://github.com/h2oai/h2o-llmstudio/blob/main/documentation/README.md) from the main readme [here](https://github.com/h2oai/h2o-llmstudio/blob/main/README.md#documentation). The docs readme gives instructions on how to edit the docs locally but it needs more content on sending in a PR and guidelines to follow. I will improve it. 

Created an [issue](https://github.com/h2oai/h2o-llmstudio/issues/331) to track.

File: CONTRIBUTING.md (hunk 1 above)
Comment (maxjeblick) [reply CONTRIBUTING.md]: Thanks for pointing this out. As we do not have guidelines written down explicitly, I've removed the comment.

**Review by pascal-pfeiffer — APPROVED** [review]
lgtm !

**Pull request merged**